### ISSUE TITLE
Crash in signed build when Me button is tapped

### ISSUE THREAD

**Issue by JavonDavis** [wordpress-mobile/WordPress-Android#14323]
### Expected behavior

App should not crash when clicking MeButton

### Actual behavior

App crashes when Me button tapped

### Steps to reproduce the behavior

1. Download signed 17.0-rc-1 build from the releases page
2. Sign in and tap the Me button in the top right from the My Site page

##### Tested on [device], Android [version], WPAndroid [version]
Google Pixel 4a, Android 11, 17.0-rc-1

Sentry report for the crash https://sentry.io/organizations/a8c/issues/1024074753/?project=1438088&query=is%3Aunresolved+user.email%3Ajavonldavis14%40gmail.com+version%3A16.9&statsPeriod=14d

**Comment by designsimply** [comment]
This is happening for me on both Pixel 3 and 4 with 17.0-rc-1 but it did not happen when I tested with 16.9-rc-3. Great catch!

(Aside: search for `29f7320469f14d5c9ba10991b9ff1342` in Sentry to see an example crash from my test user `design5279` for this case.)

**Comment by renanferrari** [comment]
Thanks for the opening this issue! I can confirm I was able to reproduce this issue and that the assumptions in #14324 are correct in that Proguard really is removing the `MeFragment`.

I just spent some time investigating this to make sure I understand what was causing this and I now believe I do.

**The problem**

Looking at the Sentry crash report, I was able to find that the crash was related to the use of `FragmentContainerView`, which immediately caught my attention because one of the changes me and @planarvoid  did as part of #14214 was to replace the use of `<fragment>` tags by [`FragmentContainerView`](https://developer.android.com/reference/androidx/fragment/app/FragmentContainerView), which has been the recommended way of directly instantiating fragments from layouts for a while now.

After some more investigation, I found [this bug report](https://issuetracker.google.com/issues/142601969) that indicated that Fragments that were being referenced only by `FragmentContainerView` were being incorrectly removed by Proguard, which confirmed our initial suspicions. 

So, to clarify what’s actually happening, if you go to the `me_activity.xml` layout, you'll see something like this:

```xml
<androidx.fragment.app.FragmentContainerView
    android:id="@+id/fragment_container"
    android:name="org.wordpress.android.ui.main.MeFragment"
    android:layout_width="match_parent"
    android:layout_height="match_parent" />
```

You can see that `MeFragment` is being referenced on the `android:name` attribute, which indicates to `FragmentContainerView` that this is the fragment that should be instantiated there. However, this is not currently being recognized by Proguard as a valid way of instantiating a fragment and since this is the only place where `MeFragment` is being instantiated, Proguard thinks it’s not being used anywhere in the app and deletes it.

I then went back into our codebase to look for other usages of `FragmentContainerView` to see if I could also cause a crash with them. I found 5 more usages of it:

<img width="300" src="https://user-images.githubusercontent.com/830056/112182361-fcc80e00-8bdb-11eb-8a10-7316c70db23b.png">

I smoke tested them all and luckily it seems that none of those are causing crashes, probably because they are also being referenced somewhere else in the app (feel free to double check on this).

**Potential solutions**

As for what we can do, I believe we have a few potential solutions:
1. Javon's fix on #14324 seems to work well and considering that `MeFragment` is the only problematic fragment at the moment, that's probably a good quick fix. That said, it won't prevent this from happening again with other fragments that might migrate from using `<fragment>` to `FragmentContainerView`.
1. There were a couple of suggested solutions for this issue in the issue tracker page for the bug I mentioned above, including:
   1. [Upgrading the Android Gradle Plugin to 4.1.0 or higher.](https://issuetracker.google.com/issues/142601969#comment12)
   1. [Forcing older Android Gradle Plugin versions to use the newer aapt2.](https://issuetracker.google.com/issues/142601969#comment14)

I'm not sure of what is the process to upgrade our Android Gradle Plugin, but assuming is not something we can do right away, I'd suggest we go ahead and merge Javon's PR first and then decide which of the solutions above works better for us.

@JavonDavis Let me know what you think!

Edit: added a paragraph to clarify what's actually happening.

**Comment by JavonDavis** [comment]
Thanks @renanferrari! Great work tracking down that bug report and writing all this up. I'll sync with the other platform engineers on what we can do as it relates to the suggested solutions. cc @wordpress-mobile/release-managers... 

> That said, it won't prevent this from happening again with other fragments that might migrate from using <fragment> to FragmentContainerView.

For now I've upgraded the PR to cover not only the MeFragment but all classes that extend Fragment, I think this should be a more wholesome solution.

**Comment by designsimply** [comment]
I just tested the steps for this in 17.7-rc-2 and the app did not crash for me when tapping the profile icon at top right on the My Site screen, so I believe the critical part of the issue was fixed when we first caught it. 👍 

@renanferrari I'd like to close this issue. Would you please create a new issue with any remaining tasks you think should still be done, if any? Then I can triage remaining issues/suggestions to see if we need to add them to our maintenance rotations or something else. 🙂 Thank you so much!

**Comment by renanferrari** [comment]
@designsimply I believe this was fixed by #14324 and I can't think of any related pending tasks. Closing it seems right to me. Thanks for taking care of it 👍
